grakn-engine/src/main/java/ai/grakn/engine/task/postprocessing/InstanceCountPostProcessor.java

```diff
@@ -0,0 +1,171 @@
+/*
+ * Grakn - A Distributed Semantic Database
+ * Copyright (C) 2016-2018 Grakn Labs Limited
+ *
+ * Grakn is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Grakn is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Grakn. If not, see <http://www.gnu.org/licenses/gpl.txt>.
+ */
+
+package ai.grakn.engine.task.postprocessing;
+
+import ai.grakn.GraknConfigKey;
+import ai.grakn.GraknTx;
+import ai.grakn.GraknTxType;
+import ai.grakn.Keyspace;
+import ai.grakn.concept.ConceptId;
+import ai.grakn.engine.GraknConfig;
+import ai.grakn.engine.factory.EngineGraknTxFactory;
+import ai.grakn.engine.lock.LockProvider;
+import ai.grakn.kb.log.CommitLog;
+import com.codahale.metrics.MetricRegistry;
+import com.codahale.metrics.Timer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.locks.Lock;
+
+import static com.codahale.metrics.MetricRegistry.name;
+
+/**
+ * <p>
+ *     Class responsible for updating the number of {@link ai.grakn.concept.Thing}s each {@link ai.grakn.concept.Type}
+ *     currently has.
+ * </p>
+ *
+ * @author Filipe Peliz Pinto Teixeira
+ */
+public class InstanceCountPostProcessor {
+    private final static Logger LOG = LoggerFactory.getLogger(InstanceCountPostProcessor.class);
+    private final RedisCountStorage redis;
+    private final GraknConfig engineConfig;
```

Comment:
We only use this config to get the sharding threshold, so I'd just pass in the sharding threshold itself.

Reply:
This is a valid case. I should not be holding a reference to GraknConfig

Reply:
you promised you would fix this one 😢

Reply:
Promise fulfilled